### ISSUE TITLE
Caching node_modules directory

### ISSUE THREAD

**Issue by roberto-butti** [Hi-Folks/gh-actions-yaml-generator#28]
Cache node_modules directory instead of downloaded packages.

```yaml
      - name: Cache node_modules directory
        uses: actions/cache@v2
        id: node_modules-cache
        with:
          path: node_modules
          key: ${{ runner.OS }}-build-${{ hashFiles('**/package-lock.json') }}

      - name: Install NPM packages
        if: steps.node_modules-cache.outputs.cache-hit != 'true'
        run: npm ci
```

**Comment by roberto-butti** [comment]
I 'm creating a branch in ordr to try to replace this one:
```yaml
    - name: Cache Node.js modules
      uses: actions/cache@v2
      with:
        # npm cache files are stored in `~/.npm` on Linux/macOS
        path: ~/.npm
        key: $@{{ runner.OS }}-node-$@{{ hashFiles('**/package-lock.json') }}
        restore-keys: |
          $@{{ runner.OS }}-node-
          $@{{ runner.OS }}-
```
this one cache downloaded packages in ./npm directory and then you need to install them via npm ci.

This issue is going to suggest to cache all node_modules directory. So in this case you can restore the dir without exectuing npm ci. THis is more fast, but i need to test a little bit to see if any side effects are raised. 👀

**Comment by roberto-butti** [comment]
merged with #33
